src/coreclr/nativeaot/System.Private.CoreLib/src/Internal/Runtime/CompilerHelpers/MathHelpers.cs

```diff
@@ -2,381 +2,329 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 
 using System;
+using System.Diagnostics;
 using System.Runtime;
 using System.Runtime.CompilerServices;
 using System.Runtime.InteropServices;
 
-using Internal.Runtime;
-
 namespace Internal.Runtime.CompilerHelpers
 {
     /// <summary>
-    /// Math helpers for generated code. The helpers marked with [RuntimeExport] and the type
-    /// itself need to be public because they constitute a public contract with the .NET Native toolchain.
+    /// Math helpers for generated code. The helpers here are referenced by the runtime.
     /// </summary>
+    [StackTraceHidden]
     internal static partial class MathHelpers
     {
-#if !TARGET_64BIT
-        //
-        // 64-bit checked multiplication for 32-bit platforms
-        //
+        private const double IntMaxValueOffset = 2147483648.0; // 2^31, int.MaxValue + 1
+        private const double UIntMaxValueOffset = 4294967296.0; // 2^32, uint.MaxValue + 1
 
-        private const string RuntimeLibrary = "*";
+        [RuntimeExport("Dbl2IntOvf")]
+        public static int Dbl2IntOvf(double value)
+        {
+            // Note that this expression also works properly for val = NaN case
+            if (value is > -IntMaxValueOffset - 1 and < IntMaxValueOffset)
+            {
+                int ret = (int)value;
+                // since no overflow can occur, the value always has to be within 1
```

Comment:
I know we have these asserts in the unmanaged impl currently, but I do not think they provide any value. The Intel PR is deleting them.